addons/website/models/website_page.py

```diff
@@ -178,6 +178,10 @@ def write(self, vals):
                             'url_to': url,
                             'website_id': website_id,
                         })
+                    # Sync website's homepage URL
+                    website = self.env['website'].get_current_website()
+                    if website.homepage_url == page.url:
```

Comment:
```suggestion
                    if website.homepage_url == website._handle_homepage_url(dict(homepage_url=page.url):
```

the homepage is preprocess in the create/write to rstrip the '/'

Reply:
The page.url already is "/"rstriped through `slugify` a few lines above so it shouldnt be a concern here?

Reply:
yes, now it is the same, but not sure that you will not update `_handle_homepage_url` in the future without change this code.
And in this case you will write on website each time.

Reply:
Ok I see what you mean, indeed

Reply:
Note that it can't just be
```suggestion
                    if website.homepage_url == website._handle_homepage_url(dict(homepage_url=page.url):
```
But has to be
```suggestion
                    page_url_normalized = {'homepage_url': page.url}
                    website._handle_homepage_url(page_url_normalized)
                    if website.homepage_url == page_url_normalized['homepage_url']:
```
`_handle_homepage_url` is just modifying the given dict

Diff: https://github.com/odoo/odoo/compare/a310ae0b40913b1820ee4fb95feb0d2331b3031e..5c82f400f28258d9f356e340688ef5fef56660e4

If it looks better like that you can r+?

Note that:
> yes, now it is the same, but not sure that you will not update _handle_homepage_url in the future without change this code.
And in this case you will write on website each time.

It's not what would happen, it's the opposite: we would not write on website anymore at all since those would not be equals. But the test would fail and we would see the bug.

So it's indeed the "correct" version the way you suggest, but it's a bit more hard to read/understand and would be protected by the test if it broke, so I am fine with both ways :+1:

Reply:
Params is given in `Call by sharing` (~ref for dict) in python so it will works, but cleaner and more explicit as you did.

```py
>>> def add(myDict, value):
...     myDict['sum'] += value
... 
>>> x = dict(sum=0)
>>> add(x, 2)
>>> x
{'sum': 2}

```

:+1: for always vs never -> in all case not what we wanted :innocent:

Reply:
:+1: